mvn clean fails in a multi-module project.

Hi, When using mvn-golang-wrapper, mvn clean fails when there are dependencies on other modules in a multi-module project. This is happening even in the examples in this repo.

The actual issue is due to the fact that dependent artifacts are not available.

The root cause seems to be due to the issue that `com.igormaznitsa.mvngolang.GolangCleanMojo` declares `requiresDependencyResolution = ResolutionScope.COMPILE`.

Actually, its not really golang issue, but more of maven and dependency management problem.

Attached an example repro.

running `mvn clean` in the parent project throws the error.

`[ERROR] Failed to execute goal on project mvn-go-test-mix-app: Could not resolve dependencies for project com.igormaznitsa:mvn-go-test-mix-app:mvn-golang:1.0.0-SNAPSHOT: The following artifacts could not be resolved: com.igormaznitsa:mvn-go-test-mix-terminal:zip:1.0.0-SNAPSHOT, com.igormaznitsa:mvn-go-test-mix-proxy:zip:1.0.0-SNAPSHOT: Could not find artifact com.igormaznitsa:mvn-go-test-mix-terminal:zip:1.0.0-SNAPSHOT -> [Help 1]`

[repro.tar.gz](https://github.com/raydac/mvn-golang/files/6757990/repro.tar.gz)

--- a/mvn-golang-wrapper/src/main/java/com/igormaznitsa/mvngolang/GolangCleanMojo.java
+++ b/mvn-golang-wrapper/src/main/java/com/igormaznitsa/mvngolang/GolangCleanMojo.java
@@ -32,7 +32,7 @@ import org.codehaus.plexus.util.FileUtils;
 /**
  * The Mojo wraps the 'clean' command.
  */
-@Mojo(name = "clean", defaultPhase = LifecyclePhase.CLEAN, threadSafe = true, requiresDependencyResolution = ResolutionScope.COMPILE)
+@Mojo(name = "clean", defaultPhase = LifecyclePhase.CLEAN, threadSafe = true)
 public class GolangCleanMojo extends AbstractGoPackageAndDependencyAwareMojo {
 
   /**

